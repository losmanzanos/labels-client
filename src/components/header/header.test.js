import renderer from "react-test-renderer";

//Smoke Test

// make React available
import React from "react";

// make the ReactDOM available, necessary for rendering the component
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";

// make the App component available
import Header from "./header";

// this is the test case
it("renders without crashing", () => {
  // first create a DOM element to render the component into
  const div = document.createElement("div");

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>,
    div
  );

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

//Snapshot

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
