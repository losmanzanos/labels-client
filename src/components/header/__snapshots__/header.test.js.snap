// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<nav
  className="Header"
>
  <div
    className="Harold"
  >
    <h1>
      Harold
    </h1>
    <div>
      <img
        alt="Harold"
        id="Harold"
        src="Harold.jpg"
      />
    </div>
  </div>
  <div
    className="Header__not-logged-in"
  >
    <a
      aria-current="page"
      className="hover active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
     | 
    <a
      aria-current={null}
      className="hover"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
     | 
    <a
      aria-current={null}
      className="hover"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
     | 
    <a
      aria-current={null}
      className="hover"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
</nav>
`;
